iml-gui/crate/src/components/breadcrumbs.rs

```diff
@@ -0,0 +1,106 @@
+use crate::{generated::css_classes::C, Msg, Page};
+use seed::{prelude::*, *};
+use std::cmp::PartialEq;
+use std::collections::LinkedList;
+
+pub struct BreadCrumbs<Crumb> {
+    crumbs: LinkedList<Crumb>,
+}
+
+impl<Crumb> Default for BreadCrumbs<Crumb> {
+    fn default() -> Self {
+        Self {
+            crumbs: LinkedList::new(),
+        }
+    }
+}
+
+impl<Crumb: PartialEq> BreadCrumbs<Crumb> {
+    pub fn iter(self: &Self) -> impl DoubleEndedIterator<Item = &Crumb> {
+        self.crumbs.iter()
+    }
+
+    pub fn clear(self: &mut Self) -> &mut Self {
+        self.crumbs.clear();
+        self
+    }
+
+    pub fn push(self: &mut Self, n: Crumb) -> &mut Self {
+        let mut new_crumbs = LinkedList::new();
+
+        while let Some(c) = self.crumbs.pop_front() {
+            if c != n {
+                new_crumbs.push_back(c);
+            } else {
+                break;
+            }
+        }
+
+        new_crumbs.push_back(n);
+        self.crumbs = new_crumbs;
+        self
+    }
+}
+
+pub fn view(bc: &BreadCrumbs<Page>) -> impl View<Msg> {
+    let mut ol = ol![class![C.justify_center, C.truncate, C.mx_4, C.rtl]];
+
+    // XXX the list has "direction: rtl" to put ellipsis to the left on overflow,
+    // XXX need to reverse the crumbs to show them in the correct left-to-right order:
```

Comment:
The comment prefix `XXX` means something, correct? For me it looks unnecessary.

Reply:
It is a placeholder for "WTF" :)

> Use XXX in a comment to flag something that is bogus but works. Use FIXME to flag something that is bogus and broken.

[[1]](https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-137265.html#395)